### PR TITLE
daniel_millan

### PR DESCRIPTION
**Pull request by drmillan**
Here are my "maybe not correct" results to exercises 1-3.

### CODE REVIEW

File: daniel_millan/session01/exercise01/cm.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: wordpress-mariadb-config
+  namespace: exercise-01
+data:
+  wordpress.log.level: --log-level=trace
```
Comment (juan131) [inline daniel_millan/session01/exercise01/cm.yaml]: Good for debugging. Please remove from final version

File: daniel_millan/session01/exercise01/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: wordpress-ingress
+  namespace: exercise-01
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
```
Comment (juan131) [inline daniel_millan/session01/exercise01/ingress.yaml]: Does it make sense to have this annotation without the complement annotations?

```
nginx.ingress.kubernetes.io/session-cookie-name
nginx.ingress.kubernetes.io/session-cookie-hash
```

File: daniel_millan/session01/exercise01/mariadb-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        - name: MARIADB_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mariadb-password
+              key: password
+        - name: MARIADB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mariadb-password
+              key: password
```
Comment (juan131) [inline daniel_millan/session01/exercise01/mariadb-deployment.yaml]: I would use different password for "root" user and the one you create for "wp"

File: daniel_millan/session01/exercise01/mariadb-svc.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  name: mariadb
+  namespace: exercise-01
+  labels:    
+    app: mariadb
+spec:
+  ports:
+    - port: 3306
+  selector:
+    app: mariadb
+  clusterIP: None
```
Comment (juan131) [inline daniel_millan/session01/exercise01/mariadb-svc.yaml]: Why did you add this?

File: daniel_millan/session01/exercise01/wordpress-deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: wordpress
+  namespace: exercise-01
+  labels:
+    app: wordpress
+    service: frontend
+spec:
+  replicas: 8
```
Comment (juan131) [inline daniel_millan/session01/exercise01/wordpress-deployment.yaml]: Why do you need so many replicas?

File: daniel_millan/session01/exercise01/wordpress-deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    metadata:
+      labels:
+        app: wordpress
+        service: frontend
+    spec:
+      containers:
+      - image: bitnami/wordpress:latest
+        name: wordpress
+        env:
+        - name: NAMI_DEBUG
```
Comment (juan131) [inline daniel_millan/session01/exercise01/wordpress-deployment.yaml]: Remove this env. variable. It's just for debugging

File: daniel_millan/session01/exercise01/wordpress-svc.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress
+  namespace: exercise-01
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
```
Comment (juan131) [inline daniel_millan/session01/exercise01/wordpress-svc.yaml]: This annotation is used on Ingress not on the service. Please remove it

File: daniel_millan/session01/exercise01/wordpress-deployment.yaml [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  template:
+    metadata:
+      labels:
+        app: wordpress
+        service: frontend
+    spec:
+      containers:
+      - image: bitnami/wordpress:latest
+        name: wordpress
+        env:
```
Comment (juan131) [inline daniel_millan/session01/exercise01/wordpress-deployment.yaml]: I'm missing some env. variables such as **WORDPRESS_USERNAME**, **WORDPRESS_PASSWORD**, **WORDPRESS_BLOG_NAME**, etc. so you can customise your WordPress site

File: daniel_millan/session01/exercise01/commands.bash [hunk 9]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+context(){
+	kubectl config set-context $(kubectl config current-context) --namespace=exercise-01
+}
+delete(){
+	kubectl delete -f .
```
Comment (juan131) [inline daniel_millan/session01/exercise01/commands.bash]: You can use: `kubectl delete ns exercise-01` to remove every object

File: daniel_millan/session01/exercise01/wordpress-deployment.yaml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+              name: wordpress-mariadb-config
+              key: database.user
+        - name: WORDPRESS_DATABASE_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mariadb-password
+              key: password
+        ports:
+        - containerPort: 80
+          name: wordpress
```
Comment (juan131) [inline daniel_millan/session01/exercise01/wordpress-deployment.yaml]: I would a name such as "http" so you can add another port (443) and name it "https"

File: daniel_millan/session01/exercise02/mariadb-deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+              key: password
+        - name: MARIADB_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mariadb-password
+              key: password
+        ports:
+        - containerPort: 3306
+          name: mariadb
+        livenessProbe:
```
Comment (juan131) [inline daniel_millan/session01/exercise02/mariadb-deployment.yaml]: Maybe not the best probe to test MariaDB.. Alternative:

```
          exec:
            command:
            - mysqladmin
            - ping
```

File: daniel_millan/session01/exercise02/mariadb-svc.yaml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: mariadb
+  namespace: exercise-02
+  labels:    
+    app: mariadb
+spec:
+  ports:
```
Comment (juan131) [inline daniel_millan/session01/exercise02/mariadb-svc.yaml]: Set a name for the port. It's a good practise

File: daniel_millan/session01/exercise02/wordpress-svc.yaml [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+metadata:
+  name: wordpress
+  namespace: exercise-02
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
+  labels:
+    app: wordpress
+spec:
+  ports:
+    - port: 80
```
Comment (juan131) [inline daniel_millan/session01/exercise02/wordpress-svc.yaml]: Again, choose a name for the port

File: daniel_millan/session01/exercise03/commands.bash [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+# STEP 2: Launch edit on nginx image
+# kubectl edit deployment nginx-ingress-controller
+
+# STEP 3: Search and replace container image to:
+# quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.10.2
+
+# STEP 4: Save file, and wait for ingress to be updated
+
+createCert(){
```
Comment (juan131) [inline daniel_millan/session01/exercise03/commands.bash]: Great!

File: daniel_millan/session01/exercise03/ingress.yaml [hunk 15]
```diff
@@ -0,0 +1,20 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: wordpress-ingress
+  namespace: exercise-03
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
+spec:
+  tls:
```
Comment (juan131) [inline daniel_millan/session01/exercise03/ingress.yaml]: Nice!

File: daniel_millan/session01/exercise03/ingress_wpadmin.yaml [hunk 16]
```diff
@@ -0,0 +1,21 @@
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: wordpress-ingress-redirect
+  namespace: exercise-03
+  annotations:
+    nginx.ingress.kubernetes.io/affinity: cookie
+    nginx.ingress.kubernetes.io/rewrite-target: /
```
Comment (juan131) [inline daniel_millan/session01/exercise03/ingress_wpadmin.yaml]: Shouldn't it redirect me to `/wp-admin`?

File: daniel_millan/session01/exercise03/tls.crt [hunk 17]
```diff
@@ -0,0 +1,21 @@
+-----BEGIN CERTIFICATE-----
```
Comment (juan131) [inline daniel_millan/session01/exercise03/tls.crt]: Remove certificate this from the PR

File: daniel_millan/session01/exercise03/tls.key [hunk 18]
```diff
@@ -0,0 +1,28 @@
+-----BEGIN PRIVATE KEY-----
```
Comment (juan131) [inline daniel_millan/session01/exercise03/tls.key]: Remove key from PR

File: daniel_millan/session01/exercise03/tls.crt (hunk 17 above)
Comment (juan131) [inline daniel_millan/session01/exercise03/tls.crt]: Remove certificate from PR

File: daniel_millan/session01/exercise01/wordpress-svc.yaml [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+    app: wordpress
+spec:
+  ports:
+    - port: 80
+  selector:
+    app: wordpress
+  type: LoadBalancer
```
Comment (juan131) [inline daniel_millan/session01/exercise01/wordpress-svc.yaml]: LoadBalancer? We don't have this type available on the Sandbox. Use `ClusterIP` since you're accessing to your site via NGINX Ingress

File: daniel_millan/session01/exercise02/wordpress-deployment.yaml [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+        - name: WORDPRESS_DATABASE_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mariadb-password
+              key: password       
+        ports:
+        - containerPort: 80
+          name: wordpress
+        livenessProbe:
+          httpGet:
```
Comment (juan131) [inline daniel_millan/session01/exercise02/wordpress-deployment.yaml]: Really nice!

File: daniel_millan/session01/exercise03/ingress_wpadmin.yaml (hunk 16 above)
Comment (drmillan) [reply daniel_millan/session01/exercise03/ingress_wpadmin.yaml]: Yes, my fault sending the PR, I have been playing with the annotation.

File: daniel_millan/session01/exercise01/wordpress-deployment.yaml (hunk 5 above)
Comment (drmillan) [reply daniel_millan/session01/exercise01/wordpress-deployment.yaml]: Playing with increasing and lowering the replica count from CLI

File: daniel_millan/session01/exercise03/wordpress-svc.yaml [hunk 21]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress-svc
+  namespace: exercise-03
+  annotations:
```
Comment (juan131) [inline daniel_millan/session01/exercise03/wordpress-svc.yaml]: You don't need to use annotation on the service definition

File: daniel_millan/session01/exercise02/wordpress-svc.yaml [hunk 22]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: wordpress
+  namespace: exercise-02
+  annotations:
```
Comment (juan131) [inline daniel_millan/session01/exercise02/wordpress-svc.yaml]: Do not use NGINX annotations on service definitions
